## Investigasi sampel data titanic berikut dengan cara :
1. Cek secara head, tail, sample, info lalu observasi apa yang bisa anda peroleh ?
2. Lakukan Statistical Summary dengan mengekstrak informasi yang didapat dari observasi anda ?
3. Cek apakah ada duplikat dan bagaimana handlenya ?
4. Cek apakah ada missing value, berapa persentasenya jika ada, dan bagaimana cara handlenya ?

## Import Libraries

In [1]:
!pip install dash pandas plotly openpyxl

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.0/8.0 MB 67.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.7/101.7 kB 7.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 228.0/228.0 kB 16.7 MB/s eta 0:00:00
  Attempting uninstall: Werkzeug
    Found existing installation: Werkzeug 3.1.3
    Uninstalling Werkzeug-3.1.3:
      Successfully uninstalled Werkzeug-3.1.3
  Attempting uninstall: Flask
    Found existing installation: Flask 3.1.0
    Uninstalling Flask-3.1.0:
      Successfully uninstalled Flask-3.1.0


In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import dash
from dash import html, dcc
import plotly.express as px
import pandas as pd
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", None)

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Load Dataset

Kode ini mengimpor data Titanic dari file Excel, menyalinnya ke variabel data, lalu mengubah nilai kolom survived dari angka (0 dan 1) menjadi label yang lebih mudah dipahami: 'Tidak Selamat' dan 'Selamat'.





In [4]:
# import data
df = pd.read_excel('/content/drive/MyDrive/EXkelas/dibimbing/DS&DA/Day 3/Assignment/titanic.xlsx')
data = df.copy()
data['survived'] = data['survived'].map({0: 'Tidak Selamat', 1: 'Selamat'})

#

# Pemeriksaan Awal

Jumlah data: 500 baris × 4 kolom
Kolom: survived, name, sex, age

👀 Head (5 teratas):
Contoh data:

Nama: Allen, Miss. Elisabeth Walton

Umur: 29

Jenis kelamin: female

Survived: 1 (selamat)

👀 Tail (5 terbawah):
Terdapat beberapa baris dengan nilai age kosong (NaN).

👀 Sample (acak):
Terlihat distribusi acak antara penumpang laki-laki/perempuan, usia muda-tua, dan status selamat/tidak.



In [5]:
print(df.head())
print(df.tail())
print(df.sample(5, random_state=42))
df.info()

   survived                                             name     sex      age
0         1                    Allen, Miss. Elisabeth Walton  female  29.0000
1         1                   Allison, Master. Hudson Trevor    male   0.9167
2         0                     Allison, Miss. Helen Loraine  female   2.0000
3         0             Allison, Mr. Hudson Joshua Creighton    male  30.0000
4         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female  25.0000
     survived                                     name     sex   age
495         1  Mallet, Mrs. Albert (Antoinette Magnin)  female  24.0
496         0        Mangiavacchi, Mr. Serafino Emilio    male   NaN
497         0               Matthews, Mr. William John    male  30.0
498         0                Maybery, Mr. Frank Hubert    male  40.0
499         0                McCrae, Mr. Arthur Gordon    male  32.0
     survived                                               name     sex   age
361         1                       Cal

#Observasi Awal

Kolom age memiliki nilai yang hilang (missing values).

Kolom survived adalah target biner (1: selamat, 0: tidak).

name kemungkinan besar unik (tapi akan dicek lebih lanjut).

Data sex terdiri dari dua kategori: male, female.

# 📊 Statistical Summary



In [6]:
len(data)

500

In [7]:
data.describe(include='all')

survived                            name   sex         age
count       500                             500   500  451.000000
unique        2                             499     2         NaN
top     Selamat  Eustis, Miss. Elizabeth Mussey  male         NaN
freq        270                               2   288         NaN
mean        NaN                             NaN   NaN   35.917775
std         NaN                             NaN   NaN   14.766454
min         NaN                             NaN   NaN    0.666700
25%         NaN                             NaN   NaN   24.000000
50%         NaN                             NaN   NaN   35.000000
75%         NaN                             NaN   NaN   47.000000
max         NaN                             NaN   NaN   80.000000

In [8]:
duplicates = data[data.duplicated(keep=False)]

In [9]:
duplicates

survived                            name     sex   age
104  Selamat  Eustis, Miss. Elizabeth Mussey  female  54.0
349  Selamat  Eustis, Miss. Elizabeth Mussey  female  54.0

Kolom survived (0 = tidak selamat, 1 = selamat):
Rata-rata: 0.54 → Sekitar 54% penumpang selamat.

Penyebaran (std): 0.5 → Distribusi seimbang.

Median: 1 → Setengah lebih selamat.

Kolom age:
Rata-rata: ~35.9 tahun

Umur termuda: ~0.67 tahun

Umur tertua: 80 tahun

Q1-Q3: 24 – 47 tahun → Mayoritas dewasa

Kolom sex:
2 kategori: male, female

Paling sering muncul: male (288 kali)

Kolom name:
499 nilai unik dari 500 → Ada 1 nama yang duplikat.

# 🔁 Cek Duplikat

In [10]:
data = data.drop_duplicates()

In [11]:
len(data.drop_duplicates()) / len(data)

1.0

❗ Cek Missing Value dan Persentasenya

Sekarang kita akan melihat apakah ada missing values. ​​

In [12]:
data.isna().sum()

survived     0
name         0
sex          0
age         49
dtype: int64

In [13]:
data.isnull().sum()

survived     0
name         0
sex          0
age         49
dtype: int64

Kolom age memiliki 451 dari 500 nilai → artinya terdapat 49 nilai kosong.

# Mengatasi missing value pada kolom age

In [14]:
# Drop duplikat
data = df.drop_duplicates()

# Isi missing value pada kolom 'age' dengan nilai rata-rata
data['age'].fillna(data['age'].mean(), inplace=True)

<ipython-input-14-bfd1030baf66>:5: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data['age'].fillna(data['age'].mean(), inplace=True)
<ipython-input-14-bfd1030baf66>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['age'].fillna(data['age'].mean(), inplace=True)


In [15]:
# Cek missing value dan persentasenya
data.isnull().sum()
(data.isnull().sum() / len(df)) * 100

survived    0.0
name        0.0
sex         0.0
age         0.0
dtype: float64

In [16]:
for column in data.columns:
    print(f"============= {column} =================")
    display(data[column].value_counts())
    print()

============= survived =================


survived
1    269
0    230
Name: count, dtype: int64


============= name =================


name
McCrae, Mr. Arthur Gordon                                                             1
Allen, Miss. Elisabeth Walton                                                         1
Allison, Master. Hudson Trevor                                                        1
Lehmann, Miss. Bertha                                                                 1
Laroche, Mrs. Joseph (Juliette Marie Louise Lafargue)                                 1
Laroche, Mr. Joseph Philippe Lemercier                                                1
Laroche, Miss. Simonne Marie Anne Andree                                              1
Laroche, Miss. Louise                                                                 1
Lamb, Mr. John Joseph                                                                 1
Lahtinen, Rev. William                                                                1
Lahtinen, Mrs. William (Anna Sylfven)                                                 1
Kvillner, Mr. Johan Henrik Johannesson                                                1
Knight, Mr. Robert J                                                                  1
Kirkland, Rev. Charles Leonard                                                        1
Kelly, Mrs. Florence "Fannie"                                                         1
Keane, Mr. Daniel                                                                     1
Keane, Miss. Nora A                                                                   1
Karnes, Mrs. J Frank (Claire Bennett)                                                 1
Kantor, Mrs. Sinai (Miriam Sternin)                                                   1
Borebank, Mr. John James                                                              1
Bonnell, Miss. Elizabeth                                                              1
Bonnell, Miss. Caroline                                                               1
Blank, Mr. Henry                                                                      1
Blackwell, Mr. Stephen Weart                                                          1
Bjornstrom-Steffansson, Mr. Mauritz Hakan                                             1
Bissette, Miss. Amelia                                                                1
Bishop, Mrs. Dickinson H (Helen Walton)                                               1
Bishop, Mr. Dickinson H                                                               1
Birnbaum, Mr. Jakob                                                                   1
Bird, Miss. Ellen                                                                     1
Bidois, Miss. Rosalie                                                                 1
Behr, Mr. Karl Howell                                                                 1
Beckwith, Mrs. Richard Leonard (Sallie Monypeny)                                      1
Beckwith, Mr. Richard Leonard                                                         1
Beattie, Mr. Thomson                                                                  1
Cardeza, Mrs. James Warburton Martinez (Charlotte Wardle Drake)                       1
Cardeza, Mr. Thomas Drake Martinez                                                    1
Candee, Mrs. Edward (Helen Churchill Hungerford)                                      1
Calderhead, Mr. Edward Pennington                                                     1
Cairns, Mr. Alexander                                                                 1
Butt, Major. Archibald Willingham                                                     1
Burns, Miss. Elizabeth Margaret                                                       1
Bucknell, Mrs. William Robert (Emma Eliza Ward)                                       1
Brown, Mrs. John Murray (Caroline Lane Lamson)                                        1
Brown, Mrs. James Joseph (Margaret Tobin)                                             1
Brewe, Dr. Arthur Jackson          


============= sex =================


sex
male      288
female    211
Name: count, dtype: int64


============= age =================


age
35.877593    49
24.000000    23
30.000000    20
36.000000    19
18.000000    14
35.000000    14
45.000000    14
42.000000    14
28.000000    12
22.000000    12
25.000000    11
21.000000    11
23.000000    11
31.000000    11
48.000000    11
34.000000    10
29.000000    10
47.000000    10
50.000000    10
39.000000     9
54.000000     9
27.000000     9
33.000000     8
55.000000     8
38.000000     8
49.000000     8
40.000000     8
19.000000     8
17.000000     7
60.000000     7
32.000000     7
26.000000     7
44.000000     6
52.000000     6
58.000000     6
37.000000     6
57.000000     5
64.000000     5
46.000000     5
51.000000     5
53.000000     4
61.000000     4
56.000000     4
43.000000     4
16.000000     4
62.000000     4
1.000000      3
41.000000     3
8.000000      3
63.000000     3
15.000000     2
65.000000     2
71.000000     2
20.000000     2
4.000000      2
6.000000      2
2.000000      1
0.916700      1
14.000000     1
11.000000     1
76.000000     1
59.000000     1
80.000000     1
28.500000     1
32.500000     1
70.000000     1
45.500000     1
13.000000     1
67.000000     1
12.000000     1
0.833300      1
0.666700      1
7.000000      1
3.000000      1
Name: count, dtype: int64

Visualisasi Data

In [17]:
df = data

# Visualisasi 1: Distribusi Penumpang Selamat vs Tidak Selamat
fig1 = px.histogram(df, x='survived', title='Distribusi Penumpang: Selamat vs Tidak Selamat', color='survived')

# Visualisasi 2: Distribusi Jenis Kelamin Penumpang
fig2 = px.histogram(df, x='sex', title='Distribusi Jenis Kelamin Penumpang', color='sex')

# Visualisasi 3: Distribusi Umur Penumpang
fig3 = px.histogram(df, x='age', nbins=30, title='Distribusi Umur Penumpang')

# Visualisasi 4: Perbandingan Survival Berdasarkan Jenis Kelamin
fig4 = px.histogram(df, x='sex', color='survived', barmode='group',
                    title='Perbandingan Jenis Kelamin dan Status Keselamatan')

# Dash app layout
app = dash.Dash(__name__)
app.title = "Dashboard Titanic"

app.layout = html.Div([
    html.H1("Dashboard Analisis Titanic", style={'textAlign': 'center'}),

    dcc.Graph(figure=fig1),
    dcc.Graph(figure=fig2),
    dcc.Graph(figure=fig3),
    dcc.Graph(figure=fig4),
])

if __name__ == '__main__':
    app.run(debug=True)

<IPython.core.display.Javascript object>